tests/abstract_test.py

```diff
@@ -240,3 +305,4 @@ def drifting_dynamics(dom):
 
 if __name__ == '__main__':
     test_abstract_the_dynamics()
+    test_abstract_the_dynamics_dual()
```

Comment:
Newline needed at end of file. An editor setting can ensure this ending whenever saving a file (e.g., [Atom](atom.io)).